### PR TITLE
feat(resources)!: enforce RFC 1035 names

### LINKED ISSUE #13047: Remove deprecations for resources whose name breaks RFC-1035
### Description

In 2.10.x we add deprecation for the resources names that break RFC-1035. And I gonna clean up in 2.12.x

xref: https://github.com/kumahq/kuma/pull/13003

### PR DESCRIPTION
**Pull request by lukidzi**
## Motivation

Names of `Mesh`, `Zone`, `MeshService`, `MeshExternalService` and `MeshMultiZoneService` end up in DNS hostnames rendered by `HostnameGenerator`, so a name that is not a valid RFC 1035 label produces an invalid hostname. Since 2.10.x such names only produced a deprecation warning (#13003) and the plan was to reject them in a later major release. Fixes #13047.

## Implementation information

Adds `validators.ValidateRFC1035Name` and calls it from the `Validate()` of the five resources, so the check runs on both the universal API server path (resource manager) and the Kubernetes admission webhook. On Kubernetes the display name is validated, which is the name of the underlying object, not the `<name>.<namespace>` core name. The corresponding RFC 1035 deprecation warnings are removed, and the `MeshService`/`MeshExternalService` name-length checks are dropped since the label rule already caps names at 63 characters — this also turns the `MeshMultiZoneService` 63-character deprecation from 2.14.x into a hard error. `KUMA_MULTIZONE_ZONE_NAME` is validated with the same rule instead of the looser 'valid DNS name, at most 63 characters', so a Zone control plane with a non-conforming name fails at startup rather than when the `Zone` resource is created on the Global control plane.

## Supporting documentation

`UPGRADE.md` documents the new rule and the rename required before upgrading. Notice that the difference between [RFC 1123](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names) and [RFC 1035](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names) is that RFC 1035 requires the name to start with an alphabetic character, which is what Kubernetes `Service` follows.

### CODE REVIEW

File: pkg/core/validators/common_validators.go [hunk 1]
```diff
@@ -292,6 +293,21 @@ func ValidateLength(path PathBuilder, maxLength int, v string) ValidationError {
...
+// ValidateRFC1035Name checks that the name is a valid RFC 1035 label. Names of
+// resources that are rendered into DNS hostnames by HostnameGenerator have to
+// conform to it. An empty name is skipped, it's rejected by the meta validation.
+func ValidateRFC1035Name(path PathBuilder, name string) ValidationError {
+	var err ValidationError
+	if name == "" {
+		return err
+	}
+	if violations := apimachineryvalidation.NameIsDNS1035Label(name, false); len(violations) > 0 {
+		err.AddViolationAt(path, strings.Join(violations, "; "))
```
Comment (lukidzi) [inline pkg/core/validators/common_validators.go]: Suggestion: Minor consistency nit — this joins errors with `; ` but multicluster.go:89 uses `, `. Not blocking, but you might want to align them.

File: pkg/core/resources/apis/mesh/mesh_validator_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+				verr := mesh.Validate()
+				// and
+				actual, err := yaml.Marshal(verr)
+
+				// then
+				Expect(err).ToNot(HaveOccurred())
+				// and
+				Expect(actual).To(MatchYAML(expected))
+			},
+			Entry("valid name", "mesh-1", "null"),
```
Comment (lukidzi) [inline pkg/core/resources/apis/mesh/mesh_validator_test.go:31]: Suggestion: Could be worth adding test entries for the length boundary — 63 chars (valid max) and 64 chars (just over). The current tests cover character-level rules but not the length cutoff.

File: pkg/core/resources/apis/system/zone_validator_test.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+			// then
+			if expectedViolation {
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("name: a DNS-1035 label must consist of lower case alphanumeric characters"))
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+		},
+		Entry("valid name", "east-1", false),
```
Comment (lukidzi) [inline pkg/core/resources/apis/system/zone_validator_test.go]: Suggestion: Similar to mesh_validator_test, consider adding a case for names >63 chars. multicluster_test.go already covers it for ZoneConfig, but not here.

File: pkg/core/validators/common_validators.go [hunk 4]
```diff
@@ -292,6 +293,21 @@ func ValidateLength(path PathBuilder, maxLength int, v string) ValidationError {
 	return err
 }
 
+// ValidateRFC1035Name checks that the name is a valid RFC 1035 label. Names of
+// resources that are rendered into DNS hostnames by HostnameGenerator have to
+// conform to it. An empty name is skipped, it's rejected by the meta validation.
+func ValidateRFC1035Name(path PathBuilder, name string) ValidationError {
```
Comment (lukidzi) [inline pkg/core/validators/common_validators.go:300]: Suggestion: The comment mentions empty names are 'rejected by the meta validation' — might help future readers to note where that happens, so they understand the validation chain.

**Review by lukidzi — COMMENTED** [review]
Automated review — DRAFT, not submitted.

Verdict: **pass**

CI: all green (15/15 checks passed)

4 non-blocking suggestions below. Solid implementation — the RFC 1035 validation is consistent across all five resources, tests cover the key cases, and UPGRADE.md is clear.

File: pkg/core/validators/common_validators.go (hunk 1 above)
Comment (lukidzi) [reply pkg/core/validators/common_validators.go]: Aligned — now joins with `, ` to match multicluster.go.

File: pkg/core/resources/apis/mesh/mesh_validator_test.go (hunk 2 above)
Comment (lukidzi) [reply pkg/core/resources/apis/mesh/mesh_validator_test.go:31]: Added entries for 63 chars (valid) and 64 chars (`must be no more than 63 characters`).

File: pkg/core/resources/apis/system/zone_validator_test.go (hunk 3 above)
Comment (lukidzi) [reply pkg/core/resources/apis/system/zone_validator_test.go]: Added 63/64-character entries; reworked the table to assert per-entry violation messages.

File: pkg/core/validators/common_validators.go (hunk 4 above)
Comment (lukidzi) [reply pkg/core/validators/common_validators.go:300]: Reworded to point at `mesh.ValidateMeta`, which rejects empty names before `Validate()` runs.

**Review by Automaat — APPROVED** [review]
